src/clj/game/core/access.clj

```diff
@@ -1160,20 +1201,31 @@
              [access-first]))
     args))
 
-(defn do-access
-  "Starts the access routines for the run's server."
-  ([state side eid server] (do-access state side eid server nil))
+(defn access-n-cards
+  "Access a specific number of cards from a server"
+  [state side eid server n]
+  (let [access-amount (num-cards-to-access state side (first server) n)]
+    (when (:run @state)
+      (swap! state assoc-in [:run :did-access] true)
+      (max-access state n))
+    (wait-for (resolve-ability state side (choose-access access-amount server {:server server}) nil nil)
+              (unregister-floating-effects state side :end-of-access)
+              (unregister-floating-events state side :end-of-access)
+              (effect-completed state side eid))))
+
+(defn breach-server
+  "Starts the breach routines for the run's server."
+  ([state side eid server] (breach-server state side eid server nil))
   ([state side eid server {:keys [no-root access-first] :as args}]
-   (wait-for (trigger-event-sync state side :pre-access (first server))
+   (system-msg state side (str "breaches " (zone->name server)))
+   (wait-for (trigger-event-sync state side :breach-server (first server))
              (let [args (clean-access-args args)
-                   access-amount (num-cards-to-access state side (first server) args)]
+                   access-amount (num-cards-to-access state side (first server) nil)]
                (turn-archives-faceup state side server)
                (when (:run @state)
                  (swap! state assoc-in [:run :did-access] true))
-               (when-not (zero? (:total access-amount))
-                 (swap! state assoc-in [:runner :register :accessed-cards] true))
                (wait-for (resolve-ability state side (choose-access access-amount server (assoc args :server server)) nil nil)
-                         (wait-for (trigger-event-sync state side :end-access-phase {:from-server (first server)})
+                         (wait-for (trigger-event-sync state side :end-breach-server {:from-server (first server)})
                                    (unregister-floating-effects state side :end-of-access)
                                    (unregister-floating-events state side :end-of-access)
-                                   (effect-completed state side eid)))))))
+                                   (effect-completed state side eid)))))))
```

Comment:
i'd prefer to not change the end of file characters, it leads to noisy diffs with no functional changes